frontend/packages/ceph-storage-plugin/src/components/create-storage-system/payloads.ts

```diff
@@ -128,6 +130,28 @@ export const labelNodes = async (nodes: WizardNodeState[]) => {
   }
 };
 
+export const taintNodes = async (nodes: WizardNodeState[]) => {
+  const patch = [
+    {
+      op: 'add',
+      path: '/spec/taints',
+      value: [ocsTaint],
+    },
+  ];
+  const requests: Promise<K8sKind>[] = [];
+  nodes.forEach((node) => {
+    const isAlreadyTainted = node.taints?.some((taint) => _.isEqual(taint, ocsTaint));
+    if (!isAlreadyTainted) {
+      requests.push(k8sPatchByName(NodeModel, node.name, null, patch));
+    }
+  });
+  try {
+    await Promise.all(requests);
+  } catch (err) {
+    throw err;
+  }
```

Comment:
Is a try block needed if all you're doing is catching and throwing the error?

Reply:
The error is thrown so it can be caught by the caller function catch

Reply:
it's basically two try catch blocks. When one can suffice.

Reply:
ack... I will remove the try-catch...